palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/SimplePalisadeService.java

```diff
@@ -78,136 +73,32 @@ public SimplePalisadeService(final AuditService auditService, final UserService
         this.resourceService = resourceService;
         this.cacheService = cacheService;
         this.executor = executor;
+        this.aggregationService = new ResultAggregationService(auditService, cacheService);
     }
 
     @Override
     public CompletableFuture<DataRequestResponse> registerDataRequest(final RegisterDataRequest request) {
         final RequestId originalRequestId = request.getId();
         LOGGER.debug("Registering data request: {}, {}", request, originalRequestId);
+
         final GetUserRequest userRequest = new GetUserRequest().userId(request.getUserId());
```

Comment:
All of this code is now very sequential. Previously, the user request policy request and resource request went off in parallel to the different services. The separation into a ResultAggregation service is good, but please can the three requests to other services happen in parallel again? CompleteableFutures are an easy way to make this happen.

Reply:
The get calls have been updated to return CompletableFuture values which needed to be updated in the service classes and the client interfaces as well. This should help to make the requests to the other services happen in parallel as expected.